spec/controllers/account/page_controller_spec.rb

```diff
@@ -58,4 +58,50 @@
       end.not_to change(Page, :count)
     end
   end
+
+  describe 'DELETE #destroy' do
+    render_views
+    before(:each) do
+      @user = FactoryBot.create(:user, :admin)
+      sign_in @user
+    end
+    let!(:company) { create :company, user_id: @user.id }
+    let!(:page) { create :page, company: company }
+
+    context 'success' do
+      subject { delete :destroy, xhr: true, params: { company_id: company.id, id: page.id } }
+
+      it 'renders template' do
+        expect(subject).to render_template(:destroy)
+      end
+
+      it 'deletes something' do
+        expect { subject }.to change(Page, :count).by(-1)
+      end
+    end
+
+    context 'error' do
+      before :each do
+        @sale = FactoryBot.create(:user, :sale, tenant_id: @user.tenant_id)
+        sign_out @user
```

Comment:
usually this means that you have bad context construction
try to change flows to avoid such things